File: 206-reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        temp1=None
        temp2=0
        while head is not None:
          temp2=head
          head=head.next
          temp2.next=temp1
          
          temp1=temp2
        return temp1

File: 206-reverse-linked-list/ReverseLinkedList.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode *nhead, *p,*q;
    p = head;
    
    if(head == NULL || head->next == NULL) return head;
    nhead = head->next;
    while(nhead!=NULL)
    {
        if(nhead->next == NULL)
        {
            nhead->next = p;
            head->next = NULL;
            return nhead;
        }
        q = nhead->next;
        nhead->next = p;
        p = nhead;
        nhead = q;
    }

   return nhead;
}
